src/core/presentation_definition.rs

```diff
@@ -263,6 +264,50 @@ impl PresentationDefinition {
 
         Ok(())
     }
+
+    /// Validate a JSON-encoded credential against the presentation definition.
+    ///
+    /// NOTE: this method accepts a generic serde_json::Value argument and checks whether
+    /// the JSON value conforms to the presentation definition's input descriptor constraint
+    /// fields.
+    ///
+    /// If the credential satisifies the presentation definition, this method will return true.
+    pub fn check_credential_validation(&self, credential: &serde_json::Value) -> bool {
+        let selector_lock = RwLock::new(jsonpath_lib::selector(credential));
+
+        self.input_descriptors().iter().any(|descriptor| {
+            descriptor
+                .constraints()
+                .fields()
+                .iter()
+                .filter_map(|field| match field.validator() {
+                    Some(Ok(validator)) => {
+                        let is_valid = field
+                            .path()
+                            .iter()
+                            .filter_map(|path| match selector_lock.try_write() {
+                                Ok(mut selector) => selector(path).ok(),
+                                _ => None,
+                            })
```

Comment:
NABD: This is all happening serially on one thread of execution, so RwLock is not needed:
```suggestion
        let mut selector = jsonpath_lib::selector(credential);

        self.input_descriptors().iter().any(|descriptor| {
            descriptor
                .constraints()
                .fields()
                .iter()
                .filter_map(|field| match field.validator() {
                    Some(Ok(validator)) => {
                        let is_valid = field
                            .path()
                            .iter()
                            .filter_map(|path| selector(path).ok())
```

Reply:
Given the mutability of `selector`, needed something here the offers inner mutability for the selector.

Reply:
On further inspection, I think the RwLock was only needed previously where I was implementing this method, but since moving into the presentation definition scope, removing the RwLock is successful. See [24395df](https://github.com/spruceid/oid4vp-rs/pull/26/commits/24395df27994c0ed120d6bb6d5ea6ca269b2778c)